examples/python/compression/client.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 the gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An example of compression on the client side with gRPC."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import grpc
+import logging
+import threading
+import time
+
+from examples import helloworld_pb2
+from examples import helloworld_pb2_grpc
+
+_DESCRIPTION = 'A client capable of compression.'
+_COMPRESSION_OPTIONS = {
+    "none": grpc.Compression.NoCompression,
+    "deflate": grpc.Compression.Deflate,
+    "gzip": grpc.Compression.Gzip,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+_SERVER_ADDRESS = 'localhost:50051'
+
+
+def main():
+    parser = argparse.ArgumentParser(description=_DESCRIPTION)
+    parser.add_argument(
+        '--channel_compression',
+        default='none',
+        nargs='?',
+        choices=_COMPRESSION_OPTIONS.keys(),
+        help='The compression method to use for the channel.')
+    parser.add_argument(
+        '--call_compression',
+        default='none',
+        nargs='?',
+        choices=_COMPRESSION_OPTIONS.keys(),
+        help='The compression method to use for an individual call.')
+    args = parser.parse_args()
+    channel_compression = _COMPRESSION_OPTIONS[args.channel_compression]
+    call_compression = _COMPRESSION_OPTIONS[args.call_compression]
+    with grpc.insecure_channel(
```

Comment:
I will prefer to encapsulate the core logic into another function. So reader can skip the part of argument parsing.